Modules/Filtering/Path/include/itkContourExtractor2DImageFilter.hxx

```diff
@@ -655,81 +525,83 @@ ContourExtractor2DImageFilter<TInputImage>::AddSegment(VertexType from, VertexTy
     // Add the contour to the end of the list and get a reference to it.
-    m_Contours.push_back(contour);
+    contours.push_back(contour);
 
     // recall that end() is an iterator to one past the back!
-    const auto newContour = --m_Contours.end();
+    const auto newContour = --contours.end();
     // add the endpoints and an iterator pointing to the contour
     // in the list to the maps.
-    m_ContourStarts.insert(VertexContourRefPair(from, newContour));
-    m_ContourEnds.insert(VertexContourRefPair(to, newContour));
+    contourStarts.insert(VertexContourRefPair(from, newContour));
+    contourEnds.insert(VertexContourRefPair(to, newContour));
   }
-  else if (newTail != m_ContourStarts.end() && newHead == m_ContourEnds.end())
+  else if (newTail != contourStarts.end() && newHead == contourEnds.end())
   {
     // Found a single contour to which the new arc should be prepended.
     const auto tail = newTail->second;
     itkAssertOrThrowMacro((tail->front() == to), "End doesn't match Beginning");
     tail->push_front(from);
     // erase the old start of this contour
-    m_ContourStarts.erase(newTail);
+    contourStarts.erase(newTail);
     // Now add the new start of this contour.
-    m_ContourStarts.insert(VertexContourRefPair(from, tail));
+    contourStarts.insert(VertexContourRefPair(from, tail));
   }
-  else if (newTail == m_ContourStarts.end() && newHead != m_ContourEnds.end())
+  else if (newTail == contourStarts.end() && newHead != contourEnds.end())
   {
     // Found a single contour to which the new arc should be appended.
     const auto head = newHead->second;
     itkAssertOrThrowMacro((head->back() == from), "Beginning doesn't match End");
     head->push_back(to);
     // erase the old end of this contour
-    m_ContourEnds.erase(newHead);
+    contourEnds.erase(newHead);
     // Now add the new start of this contour.
-    m_ContourEnds.insert(VertexContourRefPair(to, head));
+    contourEnds.insert(VertexContourRefPair(to, head));
   }
 }
 
 
 template <typename TInputImage>
 void
-ContourExtractor2DImageFilter<TInputImage>::FillOutputs()
+ContourExtractor2DImageFilter<TInputImage>::FillOutputs(ContourContainer &   contours,
+                                                        VertexToContourMap & contourStarts,
+                                                        VertexToContourMap & contourEnds)
 {
   --m_NumberLabelsRemaining;
-  if (m_NumberOutputsWritten + m_Contours.size() > m_NumberOutputsAllocated)
+  if (m_NumberOutputsWritten + contours.size() > m_NumberOutputsAllocated)
   {
     // We do not have enough capacity; increase capacity to what we need,
     // plus a guess of one contour for each unprocessed label.
-    m_NumberOutputsAllocated = m_NumberOutputsWritten + m_Contours.size() + m_NumberLabelsRemaining;
+    m_NumberOutputsAllocated = m_NumberOutputsWritten + contours.size() + m_NumberLabelsRemaining;
     this->SetNumberOfIndexedOutputs(m_NumberOutputsAllocated);
   }
```

Comment:
This should work for this single-threaded case, but be careful when we go multi-threaded.